### DISCUSSION TITLE
Opening a Nuki lock

### DISCUSSION THREAD

**Discussion by rhamblen** [rednblkx/HomeKey-ESP32#117] [Show and tell]
<img width="1421" alt="image" src="https://github.com/user-attachments/assets/6ae055a6-ab2a-4d60-80f3-e63c7e631296">

Use MQTT in 
Server being your MQTT server
Topic being “topic/homekey.auth/#”

When a valid key is selected you will receive a JSON string containing something like…(ids are made up)
endpointId: "5ABC567D888"
homekey: true
issuerId: "7ABCB1234551EFB"
Check the homekey value is true
Check the userid is the correct value
Open the Nuki lock using appropriate command (could be MQTT out or just a button if connected into node red
Send a message

**Comment by pilot1981** [comment]
If you integrated NUKI in HomeKit, I think you can use HomeKit to connect home key with NUKI lock using automation…

**Reply by rhamblen** [reply]
I think you could, but you have more capability with the nodered deconding the mqtt packet.   eg I could use my work pass (which is a regular RFID pass) to be identified and provide access.  
So I can be very specific on who and what has access AND i can set other rules, eg time of day etc. eg cleaner has access only at a specific time and day.
